ablation/search.py

```diff
@@ -0,0 +1,132 @@
+"""A script to search best configurations for the given setting."""
+import json
+import logging
+import pathlib
+import pprint
+import re
+from copy import deepcopy
+from typing import *
+
+import click
+
+HERE = pathlib.Path(__file__).parent
+RESULTS = HERE.joinpath("results")
+
+
+logger = logging.getLogger(__name__)
+
+
+def _flatten_dict(
+    dic: Any,
+    prefix: Sequence[str] = tuple(),
+) -> Iterable[Tuple[Tuple[str, ...], float]]:
+    if not isinstance(dic, dict):
+        yield prefix, dic
+        return
+
+    for k, v in dic.items():
+        yield from _flatten_dict(dic=v, prefix=prefix + (k,))
+
+
+def flatten_dict(dic: Mapping[str, Any], prefix=None):
+    if prefix is None:
+        prefix = tuple()
+    if isinstance(prefix, str):
+        prefix = (prefix,)
+    return dict(
+        (".".join(k), v)
+        for k, v in _flatten_dict(
+            dic,
+            prefix=prefix,
+        )
+    )
+
+
+def iterate_studies_from_hpo_directory(
+    directory: pathlib.Path,
+) -> Iterable[Mapping[str, Any]]:
```

Comment:
partially copied from `pykeen_report`; does not attempt to load models, and thus only requires click to be installed.